pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -208,6 +211,12 @@ func (m *Multicluster) addCluster(cluster *multicluster.Cluster) (*Controller, *
 	m.remoteKubeControllers[cluster.ID] = &kubeController{
 		Controller: kubeRegistry,
 	}
+	// localCluster may also be the "config" cluster, in an external-istiod setup.
+	localCluster := m.opts.ClusterID == cluster.ID
+	if features.EnableConfigurationDiscovery && localCluster && m.configController != nil {
+		// register filter
+		m.configController.RegisterNameSpaceDiscoveryFilter(kubeRegistry.opts.DiscoveryNamespacesFilter.Filter)
```

Comment:
I am not sure I understand this. Isn't this called per-cluster, but `m.configController` is a singleton shared by all clusters?

Reply:
It is protected by L216, so it is only registered once.

The intention to do it here is we want to reuse the `DiscoveryNamespacesFilter`

Reply:
Ah I think I missed the `localCluster` part